openproblems/tasks/_batch_integration/_common/api.py

```diff
@@ -28,12 +34,17 @@ def check_dataset(adata, do_check_pca=False, do_check_neighbors=False):
     assert adata.var_names.is_unique
     assert adata.obs_names.is_unique
 
+    assert "n_genes_pre" in adata.uns
+
     assert "organism" in adata.uns
     assert adata.uns["organism"] in ["mouse", "human"]
 
     if do_check_pca:
         assert "X_uni_pca" in adata.obsm
 
+    if do_check_hvg:
+        assert "hvg_unint" in adata.uns
```

Comment:
Check length and contents. Are there 2000 and are they valid genes. E.g. right now `var_names_make_unique` will rename some genes, so some of your HVGs may not be valid gene names